src/main/java/org/jabref/model/pdf/search/PdfSearchResults.java

```diff
@@ -26,6 +27,27 @@ public PdfSearchResults() {
         return this.searchResults;
     }
 
+    public HashMap<String, HashMap<Integer, List<SearchResult>>> getSearchResultsByPathAndPage() {
+        HashMap<String, HashMap<Integer, List<SearchResult>>> resultsByPathAndPage = new HashMap<>();
```

Comment:
This whole structure looks a bit complicated. 
A HashMap containing another hashMap which contains a List? o.O
I wonder if this could be somehow extracted to class or simplified

Reply:
I don't think introducing classes that only wrap java.util classes helps readability.
As per @tobiasdiez's suggestion we don't actually need to have the results by page any more, so we could remove one level of HashMaps.